src/s3path.jl

```diff
@@ -187,7 +187,11 @@ function FilePathsBase.parents(fp::S3Path)
     end
 end
 
-FilePathsBase.exists(fp::S3Path) = s3_exists(fp.config, fp.bucket, fp.key; version=fp.version)
+# Use `fp.config` unless it is nothing; in that case, get the latest `global_aws_config`
+get_config(fp::S3Path) = @something(fp.config, global_aws_config())
```

Comment:
I'll mention you could use `getproperty` to make this change more seamless but I think a function is the right call personally